server/http/restapi/asset.go

```diff
@@ -0,0 +1,45 @@
+package restapi
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost-plugin-apps/server/api"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+	"github.com/pkg/errors"
+)
+
+func (a *restapi) assetGet(w http.ResponseWriter, req *http.Request) {
+	actingUserID := req.Header.Get("Mattermost-User-Id")
+	if actingUserID == "" {
+		httputils.WriteUnauthorizedError(w, errors.New("user not logged in"))
+		return
+	}
+
+	sessionID := req.Header.Get("MM_SESSION_ID")
+	if sessionID == "" {
+		httputils.WriteUnauthorizedError(w, errors.New("no user session"))
+		return
+	}
+
+	_, err := a.api.Mattermost.Session.Get(sessionID)
```

Comment:
Why are we doing the session check? I thought it comes into the plugin already validated?

Reply:
You are correct